main.go

```diff
@@ -121,6 +130,50 @@ func main() {
 					os.Exit(1)
 				}
 			}
+		} else if module, _ := tfconfig.LoadModule(dir); len(module.RequiredCore) >= 1 && len(args) == 0 { //if there is a .tfswitchrc file, and no commmand line arguments
+			tfversion := ""
+
+			// we skip duplicated definitions and use only first one
+			tfconstraint := module.RequiredCore[0]
+			tflist, _ := lib.GetTFList(hashiURL, true)
+			fmt.Printf("Reading required version from terraform code, constraint: %s\n", tfconstraint)
+
+			c, err := semver.NewConstraint(tfconstraint)
+			if err != nil {
+				fmt.Println("Error parsing constraint:", err)
+				os.Exit(1)
+			}
+			vs := make([]*semver.Version, len(tflist))
+			for i, r := range tflist {
+				v, err := semver.NewVersion(r)
+				if err != nil {
+					fmt.Printf("Error parsing version: %s", err)
+					os.Exit(1)
+				}
+
+				vs[i] = v
+			}
+
+			sort.Sort(sort.Reverse(semver.Collection(vs)))
+
+			for _, element := range vs {
+				// Validate a version against a constraint.
+				if c.Check(element) {
+					tfversion = string(element.String())
+					fmt.Printf("Matched version: %s\n", tfversion)
+
+					if lib.ValidVersionFormat(tfversion) {
+						lib.Install(string(tfversion), *custBinPath)
+						break
+					} else {
+						fmt.Println("Invalid terraform version format. Format should be #.#.# or #.#.#-@# where # is numbers and @ is word characters. For example, 0.11.7 and 0.11.9-beta1 are valid versions")
+						os.Exit(1)
+					}
+				}
+			}
+
+			fmt.Println("No version found to match constraint.")
+			os.Exit(1)
```

Comment:
But now, correct flow will also print this msg and exit with code 1. One way of doing this correctly will be to declare `isVersionFound := false` before the loop, set it `true` before the `break` and check it after the loop.

Reply:
The break is not necessary because the install function(**lib.Install(string(tfversion), *custBinPath)**) will exit if it is successful or fail if it is unsuccessful:
 For loop:
```
for _, element := range vs {
				// Validate a version against a constraint.
				if c.Check(element) {
					tfversion = string(element.String())
					fmt.Printf("Matched version: %s\n", tfversion)

					if lib.ValidVersionFormat(tfversion) {
						lib.Install(string(tfversion), *custBinPath) //will exit if it is successful or fail if it is unsuccessful
						break // Not needed
					} else {
						fmt.Println("Invalid terraform version format. Format should be #.#.# or #.#.#-@# where # is numbers and @ is word characters. For example, 0.11.7 and 0.11.9-beta1 are valid versions")
						os.Exit(1)
					}
				}
			}
```

See the install function:
```
func Install(tfversion string, binPath string) {
  .  
  os.Exit(1) // if something is unsuccessful 
  .  
  . 
  os.Exit(0)// if all is good
} 
```